code/obj/machinery/computer/buildandrepair.dm

```diff
@@ -223,7 +224,8 @@
 			if (isscrewingtool(P))
 				playsound(src.loc, "sound/items/Screwdriver.ogg", 50, 1)
 				boutput(user, "<span class='notice'>You connect the monitor.</span>")
-				var/B = new src.circuit.computertype ( src.loc )
+				var/obj/B = new src.circuit.computertype ( src.loc )
+				B.set_dir(src.dir)
 				if (circuit.powernet) B:powernet = circuit.powernet
 				if (circuit.id) B:id = circuit.id
 				if (circuit.records) B:records = circuit.records
```

Comment:
Why not `var/obj/machinery/computer/B = ...`? Then we can remove those gross `:`. Do a `null` check I guess before doing the other stuff, in case for some reason `computertype` isn't one of those.

Reply:
Done